Core/clim-basic/text-formatting.lisp

```diff
@@ -128,3 +128,88 @@
+   (alb :accessor after-line-break :initarg :after-line-break
+        :documentation "Function accepting stream to call after the line break.")
+   (ilb :accessor after-line-break-initially :initarg :after-line-break-initially
+        :documentation "Flag whenever we execture alb on fresh newlines.")
+   (slb :accessor after-line-break-subsequent :initarg :after-line-break-subsequent
+        :documentation "Flag whenever we execture alb on soft newlines.")
+   (gfs :reader graphics-state :initarg :gfs))
+  (:default-initargs :line-break-strategy t
+                     :gfs (make-instance '%filling-output-graphics-state)
+                     :after-line-break nil
+                     :after-line-break-initially nil
+                     :after-line-break-subsequent t))
+
+(defmacro filling-output ((stream &key
+                                  (fill-width ''(80 :character))
+                                  break-characters
+				  after-line-break
+                                  after-line-break-initially
+                                  (after-line-break-subsequent t))
+			  &body body)
+  (setq stream (stream-designator-symbol stream '*standard-output*))
+  `(with-temporary-margins (,stream :right (list :absolute ,fill-width))
+     (letf (((stream-end-of-line-action ,stream) :wrap*)
+            ((line-break-strategy ,stream) ,break-characters)
+            ,@(when after-line-break
+                `(((after-line-break ,stream) ,after-line-break)
+                  ((after-line-break-initially ,stream) ,after-line-break-initially)
+                  ((after-line-break-subsequent ,stream) ,after-line-break-subsequent)
+                  ((graphics-state-ink (graphics-state ,stream)) (medium-ink ,stream))
+                  ((graphics-state-text-style (graphics-state ,stream)) (medium-text-style ,stream)))))
+       ,(when (and after-line-break after-line-break-initially)
+          `(etypecase (after-line-break ,stream)
+             (string   (write-string (after-line-break ,stream) ,stream))
+             (function (funcall (after-line-break ,stream) ,stream nil))))
+       ,@body)))
+
+(defmacro indenting-output ((stream indent &key (move-cursor t)) &body body)
+  (setq stream (stream-designator-symbol stream '*standard-output*))
+  `(with-temporary-margins (,stream :left (list :absolute ,indent)
+                                    :move-cursor ,move-cursor)
+     ,@body))
+
+
+;;; formatting functions
+(defun format-textual-list (sequence printer
+                            &key stream separator conjunction
+                              suppress-separator-before-conjunction
+                              suppress-space-after-conjunction)
+  "Outputs the SEQUENCE of items as a \"textual list\" into
+STREAM. PRINTER is a function of an item and a stream. Between each
+two items the string SEPARATOR is placed. If the string CONJUCTION is
+supplied, it is placed before the last item.
```

Comment:
Shouldn't the docstring document the suppress keywords as well?

Reply:
This is simply copied from clim-core/text-formatting.lisp.